openmdao/core/group.py

```diff
@@ -1419,9 +1420,20 @@ def _setup_connections(self):
                     if np.prod(src_indices.shape) == 0:
                         continue
 
+                    if is_slice and src_indices.size == np.prod(in_shape):
```

Comment:
could do this in one line: `flat_array_slice_check = is_slice and src_indices.size == np.prod(in_shape)`